tests/common/test_utils.rs

```diff
@@ -0,0 +1,34 @@
+use std::fmt::Debug;
+use std::future::Future;
+use std::ops::FnMut;
+use tokio::time::{sleep, Duration};
+use tonic_lnd::tonic::{Response, Status};
+
+// If a grpc call returns an error, retry_async will retry the grpc function call in case lnd is still in
+// the process of starting up. A note on implementation: We can't pass in a future directly to a function
+// because futures cannot be cloned/copied in order to retry the future. Instead retry_async takes in an
+// async closure that is able to "copy" the function for us so we can call it multiple times for retries.
+pub(crate) async fn retry_async<F, Fut, D>(mut f: F, func_name: String) -> Result<D, ()>
```

Comment:
Reviewing commit by commit, so this question may be answered for me but: do we need this retry anywhere except when LND is starting up? If that's the case, LND does provide a [API](https://lightning.engineering/api-docs/category/state-service) that notifies when it's ready (makes for robust itests).

Because I'd think that not being able to make regular API calls maybe we do want the test to fail?